ros_cross_compile/ros_cross_compile.py

```diff
@@ -193,7 +197,7 @@ def main():
     """Start the cross-compilation workflow."""
     args = parse_args(sys.argv[1:])
     ros_workspace_dir = _resolve_ros_workspace(args.ros_workspace)
-    data_collector = DataCollector()
+    data_collector = DataCollector(args.print_metrics)
```

Comment:
Instead of passing this information to the object, we could probably just put another line below `data_writer.write`. The `data_writer`, actually, could even be the thing responsible for printing as well (makes sense for responsibility division) - then it might look like `data_writer.write(data_collector, also_write_stdout=args.print_metrics)`